src/sync/sync_config.hpp

```diff
@@ -100,6 +104,7 @@ struct SyncConfig {
     SyncSessionStopPolicy stop_policy;
     std::function<SyncBindSessionHandler> bind_session_handler;
     std::function<SyncSessionErrorHandler> error_handler;
+    std::shared_ptr<ChangesetTransformer> transformer;
```

Comment:
That doesn't work unfortunately. So we have an object lifetime issue if the `sync::Session` outlives the `SharedRealm`. If we stored the `ChangesetCooker` as a `shared_ptr` that would resolve this. Is that possible?